### PR TITLE
V1.5 documentation

### PR DESCRIPTION
**Pull request by yapni**
What's new:
- Added an intro and some preface to DG
- Updated the favourites and dashboard documentation in UG and DG (including manual testing)

### CODE REVIEW

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -14,25 +14,45 @@ endif::[]
 
 By: `Team W09-B3`      Since: `Mar 2018`      Licence: `MIT`
 
+== Introduction
+
+=== About Codeducator
```
Comment (demitycho) [inline docs/DeveloperGuide.adoc]: Good intro! I think this succinctly captures the gist of our app and communicates very well to potential developers

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -370,16 +394,21 @@ _{Explain here how the data encryption feature will be implemented}_
...
 === Favourites feature
+
+The favourites feature allows users to remember/mark a student by adding them to favourites.
+
 ==== Current Implementation
 
 To facilitate the favourite/unfavourite feature, an association with a new `Favourite` class is added to the `Student` class:
 
+.Structure of the atrributes of a `Student` in the Model component. The diagram shows that the `Student` class is associated with the `Favourite` class.
 image::StudentWithPLFeature.png[width="500"]
```
Comment (demitycho) [inline docs/DeveloperGuide.adoc]: Reviewed the file. It's definitely more accurate model diagram

File: docs/DeveloperGuide.adoc (hunk 1 above)
Comment (chweeee) [reply docs/DeveloperGuide.adoc]: The documentation for `favourites` and `dashboard` is more readable and clear. Good job!

File: docs/DeveloperGuide.adoc [hunk 3]
```diff
@@ -370,16 +394,21 @@ _{Explain here how the data encryption feature will be implemented}_
...
 
 Since the implementation of the favourite and unfavourite command are similar, we will describe the implementation of the favourite command only.
 
 The following sequence diagram shows how the favourite command works:
 
+.Sequence diagram for the favourite command
 image::FavouriteCommandSequenceDiagram.png[width="800"]
```
Comment (demitycho) [inline docs/DeveloperGuide.adoc]: Good representation of the sequence digram for FavouriteCommand

File: docs/DeveloperGuide.adoc [hunk 4]
```diff
@@ -1224,3 +1258,193 @@ _{ more test cases ... }_
 .. _{explain how to simulate a missing/corrupted file and the expected behavior}_
 
 _{ more test cases ... }_
+
+//tag::favTest[]
```
Comment (demitycho) [inline docs/DeveloperGuide.adoc]: Seems like all test cases are covered with your manual testing test cases.

File: docs/DeveloperGuide.adoc [hunk 5]
```diff
@@ -1224,3 +1258,193 @@ _{ more test cases ... }_
...
+.. Prerequisites: List only students in favourites using the `list -f` command and there are multiple students in the list.
+.. Test case: `unfav 1` +
+Expected:
+* First student contact in the favourites list will be removed.
+* The name of the first student contact will be shown in the result box.
+* Timestamp in the status bar is updated.
+
+//end::favTest[]
+
+//tag::dashboardTest[]
```
Comment (demitycho) [inline docs/DeveloperGuide.adoc]: Very extensive testing as well

File: docs/UserGuide.adoc [hunk 6]
```diff
@@ -202,6 +203,37 @@ List only all student that you added to favourites
 * `list` +
 List all students
 
+*Steps taken to view all students in the address book*: +
```
Comment (demitycho) [inline docs/UserGuide.adoc]: Thanks for the screenshots. Much clearer instructions!

File: docs/UserGuide.adoc [hunk 7]
```diff
@@ -469,134 +541,246 @@ You must first be logged in to sync your data
 
 Format: `sync`
 
-=== Dashboard
 // tag::dashboard[]
-==== Show the dashboard of a student: `showDB`
+=== Dashboard
```
Comment (demitycho) [inline docs/UserGuide.adoc]: Seems like a good explanation of the Dashboard feature.

File: docs/UserGuide.adoc [hunk 8]
```diff
@@ -469,134 +541,246 @@ You must first be logged in to sync your data
...
+* Milestones: +
+Each dashboard will have a list of milestones. A milestone signifies a major step in the learning progress of your student.
+Codeducator allows you to create and add milestones in your students' dashboards so that you can keep track of learning objectives you have set for them.
+
+* Tasks: +
+Each milestone can contain a list of tasks. A task signifies a piece of work to be done to meet the milestone's objective.
+Codeducator allows you to create and add tasks to each milestone so that you can keep track of what has already been done and what still needs to be done to reach the milestone.
+
+.An example of a student's dashboard
+image::dashboardScreenshot.png[width="800"]
```
Comment (demitycho) [inline docs/UserGuide.adoc]: The annotation helps with understanding your feature!

File: docs/UserGuide.adoc [hunk 9]
```diff
@@ -469,134 +541,246 @@ You must first be logged in to sync your data
...
+===== Steps taken
+
+*Step 1*: First, find the student whose dashboard you wish to view using the `list` command (See <<Listing all students : `list`>>).
+
+*Step 2*: Once you have found the student, type `showDB` into the command box, followed by the `STUDENT_INDEX` of the student in the list. Press kbd:[Enter] to execute it.
+
+.Entering the `showDB` command followed by the `STUDENT_INDEX` of the student whose dashboard you wish to view
+image::showDB1_screenshot.png[width="800"]
+
+*Step 3*: You have succeeded in viewing the student's dashboard if you see the message "Selected Dashboard of Student: `STUDENT_INDEX` " in the result box, with the dashboard containing the student's name appearing on the right panel.
```
Comment (demitycho) [inline docs/UserGuide.adoc]: Perhaps use a more conversational tone?

File: docs/UserGuide.adoc [hunk 10]
```diff
@@ -469,134 +541,246 @@ You must first be logged in to sync your data
...
+The `MILESTONE_DUE_DATE` provided must have a valid calendar date and follows the specified format. Otherwise, an error message shown in the figure below will be displayed in the result box at *Step 3*. +
 
-==== Delete a milestone from a student's dashboard: `deleteMS`
+.Error message displayed when an invalid `MILESTONE_DUE_DATE` is entered
+image::addMS4_screenshot.png[width="800"]
+
+
+==== Deleting a milestone from a student's dashboard: `deleteMS`
+
+If you wish to remove a milestone you no longer need from a student's dashboard, you can use the `deleteMS` command. +
```
Comment (demitycho) [inline docs/UserGuide.adoc]: I'd say use a more conversational tone

File: docs/DeveloperGuide.adoc (hunk 4 above)
Comment (yapni) [reply docs/DeveloperGuide.adoc]: Hmm, I think we can leave it up to the testers to vary their test inputs. Maybe it is best if we just provide suggestions to guide them on how to test.

**Pull request merged**